Datarequest

aucuparia opened the pull request.

aucuparia committed:
Replace placeholder data download instructions

aucuparia committed:
Small code cleanup

aucuparia committed:
Typo (YDA-3931)

aucuparia committed:
Typo (YDA-3932)

aucuparia committed:
Typo (YDA-3941)

aucuparia committed:
Add small remark for reviewer (YDA-3967)

aucuparia committed:
Clarify to whom the data manager review is addressed (YDA-3966)

aucuparia committed:
Add feedback for researcher to detail page of resubmitable or rejected data requests (YDA-3945)

aucuparia committed:
Typos

aucuparia committed:
flake8 fixes

lwesterhof reviewed (approved):
LGTM

aucuparia committed:
Add test for new datarequest_feedback_get API call (YDA-3945)

aucuparia committed:
Split Name field into subfields (YDA-3936)

aucuparia committed:
Typo

lwesterhof commented on datarequest/schemas/youth-0/datarequest/schema.json:
```diff
@@ -29,9 +29,13 @@
             "type": "object",
             "properties": {
-              "name": {
+              "first_name": {
```
In default-1 we use first name as label, but the field is named given_name. To be consistent with DataCite. And Given name and Family Name may also be a nicer combination.

aucuparia replied:
Agreed. Fixed.

aucuparia committed:
Split Name field into subfields (YDA-3936; additional change)

aucuparia committed:
Add email address validation (YDA-3398)

aucuparia committed:
Enlarge feedback and remark fields (YDA-3965)

aucuparia committed:
Deduplicate resubmit and reject email code

aucuparia committed:
Hide irrelevant field for data assessment only data requests (YDA-3925)

aucuparia committed:
Remove obsolete field

aucuparia committed:
Remove data lock notification field (YDA-3933)

aucuparia committed:
Make hypotheses field mandatory (YDA-3929)

aucuparia committed:
Cosmetics

aucuparia committed:
Bugfix: correct for new datarequest schema structure introduced by YDA-3925

aucuparia committed:
Bugfix: correct for splitting of Name field introduced by YDA-3936

aucuparia committed:
Remove multiple contact persons functionality (YDA-3962)

aucuparia committed:
Typo

aucuparia committed:
Fix lint error

aucuparia committed:
Add link to Data Access Protocol (YDA-3930)

aucuparia committed:
Allow BoD member to download signed DTA (YDA-3951)

aucuparia committed:
Direct objections to rejection to BoD member instead of datamanager (YDA-3944)

aucuparia committed:
Do not allow BoD members and datamanagers to submit datarequests (YDA-3923)

aucuparia committed:
Code polishing

aucuparia committed:
Remove requester email field from form (YDA-3947)

aucuparia committed:
Clarify assignment action (YDA-3938)

aucuparia committed:
Shortened review process for data assessment only data requests (YDA-4004)

aucuparia committed:
Fix lint errors

aucuparia committed:
Keep original filename of (signed) DTA (YDA-3948, WIP)

aucuparia committed:
Fix lint errors

aucuparia committed:
Log timestamps of status transitions (YDA-3747; WIP)

aucuparia committed:
Hide irrelevant field for data assessment only requests (YDA-3925, additional change)

aucuparia committed:
Add save as draft button (YDA-3394)

Commit:
Rename bodmember role to project manager (YDA-4029)

Commit:
Typos and code polishing

Commit:
Add contribution checks (YDA-4028)

Commit:
Code cleanup and minor bugfix

Commit:
Update data request schema (YDA-4038)

Commit:
Keep a provenance log (YDA-3747)

Commit:
Update data request schema (YDA-4038, bugfix)

Commit:
Keep a provenance log (YDA-3747, additional commit)

Commit:
Permissions clean-up (YDA-3948, YDA-3952, YDA-3956)

Commit:
Code polishing

Commit:
Upload supplementary material (YDA-3957)

Commit:
Add newly added forms to review pages (where missing)

Commit:
Small textual improvement

Commit:
Add missing request_id parameter type conversions

Commit:
Simplify datarequest_is_owner

Commit:
Deduplicate and simplify user role checking

Commit:
Split DTA upload permission functions into separate functions for DTA and signed DTA

Commit:
Deduplicate and simplify permission checks

Commit:
Add missing / remove obsolete permission checks

Commit:
Small bugfix

Commit:
Add missing Docstrings

Commit:
Code polishing

Commit:
flake8 error fixes

lwesterhof reviewed (changes requested):
Several new API have been introduced, is it possible to add some API tests for these?

Would also be interesting to have one complete happy flow of the module as UI test.

Commit:
Aggregate user role checks in single API call

Commit:
Update datarequest API tests

aucuparia commented:
> Several new API have been introduced, is it possible to add some API tests for these?

Done.

Commit:
Small documentation update and bugfix

Commit:
Simplify datarequest schema installation

There is no need to not overwrite already present schemas, as there is only a single datarequest schema collection that will not be changed
